src/bluesearch/database/topic.py

```diff
@@ -261,3 +256,32 @@ def _parse_mesh_from_pubmed(mesh_headings: Iterable[Element]) -> list[dict]:
             meshs.append({"descriptor": descriptor_name, "qualifiers": qualifiers_name})
 
     return meshs
+
+
+# PMC source
+def get_topics_for_pmc_article(
+    pmc_path: pathlib.Path | str,
+) -> List[str] | None:
+    """Extract journal topics of a PMC article."""
+    journal_topics = None
+
+    # Determine journal title
+    parser = JATSXMLParser(pmc_path)
+    nlm_ta = parser.content.find(
+        "./front/journal-meta/journal-id[@journal-id-type='nlm-ta']"
+    )
+    if nlm_ta is None:
+        logger.error(f"No NLM Title Abbreviation found for {pmc_path}")
+        return journal_topics
+
+    nlm_ta = nlm_ta.text
+    logger.info(f"Journal Title Abbreviation: {nlm_ta}")
+    journal_meshes = request_mesh_from_nlm_ta(nlm_ta)
+
+    if journal_meshes is not None:
+        journal_topics = []
+        for mesh in journal_meshes:
+            for descriptor in mesh["descriptor"]:
+                journal_topics.append(descriptor["name"])
+
+    return journal_topics
```

Comment:
I'd like to suggest a small refactoring that IMO make it more transparent when `None` is returned:
```suggestion
    # Determine journal title
    parser = JATSXMLParser(pmc_path)
    nlm_ta = parser.content.find(
        "./front/journal-meta/journal-id[@journal-id-type='nlm-ta']"
    )
    if nlm_ta is None:
        logger.error(f"No NLM Title Abbreviation found for {pmc_path}")
        return None

    nlm_ta = nlm_ta.text
    logger.info(f"Journal Title Abbreviation: {nlm_ta}")
    journal_meshes = request_mesh_from_nlm_ta(nlm_ta)
    if journal_meshes is None:
        return None

    journal_topics = []
    for mesh in journal_meshes:
        for descriptor in mesh["descriptor"]:
            journal_topics.append(descriptor["name"])

    return journal_topics
```